homework-submission/my-project/src/App.js

```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react';
+import './App.css';
+import Post from './Post';
+import Data from './Data';
+import Header from './Header';
+
+
+class App extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      todoData: Data
+    }
+    this.changeHandel = this.changeHandel.bind(this)
+  }
+
+  changeHandel(id) {
+    this.setState(prevState => {
+      const updatedData = prevState.todoData.map(todo => {
+        if (todo.id === id) {
+          todo.done = !todo.done;
+        }
+        return todo
+      })
+      return {
+        todoData: updatedData
+      }
+    })
+  }
+
```

Comment:
The loop here is costly, because you are looping over all the data to find one item which you want to change.
and you have in line 33 a key in the Map loop, so you can pass this key as a prop to the < Post > component and use it to find the item

Reply:
Sorry, I did't get it

Reply:
you don't need to make a loop because you have the index of the item in your array that mean you can reach the item directly

Reply:
Sorry for being late. Done